File: C#/GnomeSort.cs
using System;

class GnomeSort
{
    public static void Sort(int[] arr)
    {
        int i = 0;
        while (i < arr.Length)
        {
            if (i == 0 || arr[i] >= arr[i - 1])
                i++;
            else
            {
                int temp = arr[i];
                arr[i] = arr[i - 1];
                arr[i - 1] = temp;
                i--;
            }
        }
    }

    public static void Main()
    {
        int[] arr = { 34, 2, 10, -9 };
        Sort(arr);
        Console.WriteLine(string.Join(" ", arr));
    }
}

File: GnomeSort.java
public class GnomeSort {
    public static void gnomeSort(int[] arr) {
        int index = 0;
        int n = arr.length;

        while (index < n) {
            if (index == 0 || arr[index] >= arr[index - 1]) {
                index++;
            } else {
                int temp = arr[index];
                arr[index] = arr[index - 1];
                arr[index - 1] = temp;
                index--;
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {34, 2, 10, -9, 6};
        gnomeSort(arr);
        System.out.println("Sorted array:");
        for (int i : arr) {
            System.out.print(i + " ");
        }
    }
}